packages/astro/src/core/compile/compile.ts

```diff
@@ -42,7 +35,7 @@ export async function compile({
 	// Because `@astrojs/compiler` can't return the dependencies for each style transformed,
 	// we need to use an external array to track the dependencies whenever preprocessing is called,
 	// and we'll rebuild the final `css` result after transformation.
-	const cssDeps: CompileCssResult['dependencies'][] = [];
+	const cssPartialCompileResults: PartialCompileCssResult[] = [];
```

Comment:
Note: I refactored this from `string[][]` to `{ ... dependencies: string[] }[]` because I needed to keep an extra `isGlobal` metadata whenever preprocessing is called.